### ISSUE TITLE
Suggesting that by default a Heatmap has no annotations with the Matplotlib backend

### ISSUE THREAD

**Issue by maximlt** [holoviz/holoviews#5108]
Annotations don't look good in a Heatmap with a large amount of data and those that are rendered in the high values range are barely visible with the default color map. Heatmaps generated with the Bokeh backend don't have annotations by default.

See this example taken from the reference gallery:
![image](https://user-images.githubusercontent.com/35924738/140487561-3cdb38ea-8694-4328-ab9f-bfddaf1dd010.png)

**Comment by jbednar** [comment]
Fine by me!

**Comment by jlstevens** [comment]
I agree this should be trivial for someone to change but note that this affects backwards compatibility. That means this is suitable for 2.0 but should *not* go into a hotfix release.

**Comment by prj99** [comment]
@maximlt may I work with this issue

**Comment by maximlt** [comment]
Yes please @prj99! :)
